kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -541,6 +545,17 @@ public String convert(boolean heatCoolEq) {
         return sb.toString();
     }
 
+    private boolean isRealHook(Att att) {
+      String hook = att.get(Attribute.HOOK_KEY);
+      if (hook.startsWith("ARRAY.")) {
+        return false;
+      }
+      if (options.hookNamespaces.stream().anyMatch(ns -> hook.startsWith(ns + "."))) {
+        return true;
+      }
+      return Hooks.namespaces.stream().anyMatch(ns -> hook.startsWith(ns + "."));
+    }
```

Comment:
I guess these two could be combined with some sort of `Set.union(Hooks.namspaces, options.hookNamespaces).steam()....`

Up to you.